accepted/2021/runtime-security-mitigations.md

```diff
@@ -0,0 +1,213 @@
+
+**Table of Contents**
+
+- [Anatomy of an exploit](#anatomy-of-an-exploit)
+- [Executable Space Protection](#executable-space-protection)
+  - [W^X](#wx)
+  - [Protect Intermediate Language (IL) Code](#protect-intermediate-language-il-code)
+  - [Address Space Layout Randomization (ASLR)](#address-space-layout-randomization-aslr)
+  - [Disallow Runtime Code Generation](#disallow-runtime-code-generation)
+  - [Out-of-process Jit-In-Time Compiler (JIT)](#out-of-process-jit-in-time-compiler-jit)
+- [Control-flow Integrity](#control-flow-integrity)
+  - [Intel Control-flow Enforcement Technology (CET)](#intel-control-flow-enforcement-technology-cet)
+    - [Shadow Stack](#shadow-stack)
+    - [Indirect Branch Tracking (IBT)](#indirect-branch-tracking-ibt)
+  - [Windows Control Flow Guard (CFG)](#windows-control-flow-guard-cfg)
+  - [ARM Pointer Authentication (PA)](#arm-pointer-authentication-pa)
+  - [Stack Buffer Overflow Protection](#stack-buffer-overflow-protection)
+- [Language, Libraries and Runtime Features](#language-libraries-and-runtime-features)
+  - [Unsafe API and C# language constructs](#unsafe-api-and-c-language-constructs)
+  - [Object and Array Pooling](#object-and-array-pooling)
+  - [Secrets in Presence of GC Heap Compaction](#secrets-in-presence-of-gc-heap-compaction)
+  - [Dangerous Deserialization](#dangerous-deserialization)
+- [Looking forward](#looking-forward)
+  - [Plan for .NET 6](#plan-for-net-6)
+- [Appendix](#appendix)
+  - [Timing Side Channel Attacks](#timing-side-channel-attacks)
+  - [Confidential Computing](#confidential-computing)
+
+# Anatomy of an exploit
+
+Memory security vulnerability starts with a memory safety violation. That includes bugs such as writing outside of an array's bounds or using a pointer after freeing the associated object. When an attacker can influence the pointer or data involved, they have at least some control of the program's memory (this is referred to as a write-what-where primitive). Once an attacker has a write-what-where primitive, they may be able to use that to change the program's execution to do something malicious.
+
+The security mitigations prevent or reduce probability of each step in this sequence.
+
+The mitigations for memory safety violations must be typically applied to all code running in a process to be fully effective. The exploits often combine memory safety violations and code from multiple modules in the process. For example, memory safety violation in C/C++ module and code generated by .NET runtime JIT can be combined together to produce an exploit. Thus, missing mitigations in .NET modules can allow exploiting vulnerabilities in C/C++ modules that would be hard or impossible to exploit in isolation, and vice versa.
+
+# Executable Space Protection
+
+_Known as Data Execution Prevention (DEP) on Windows_
```

Comment:
Maybe just change the title to Executable Space Protection / Data Execution Prevention rather than have the italicized sentence?